### PR TITLE
p3x-onenote: 2020.10.111 -> 2022.4.127

### PR DESCRIPTION
**Pull request by tiagolobocastro**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

i386-linux AppImage is no longer released

### CODE REVIEW

**Review by fufexan — APPROVED** [review]

**Comment by bobby285271** [comment]
> i386-linux AppImage is no longer released

Actually I think meta.platforms should reflect the change...
Can you also set meta.platforms as `[ "aarch64-linux" "armv7l-linux" "x86_64-linux" ]`?

**Comment by LegitMagic** [comment]
p3x-onenote has a new update [v2022.10.109](https://github.com/patrikx3/onenote/releases/tag/v2022.10.109).
According to the change log, it only seems to be a minor update, so I don't expect anything will break, however I have not tried it.

File: pkgs/applications/office/p3x-onenote/default.nix [hunk 1]
```diff
@@ -1,23 +1,19 @@
 { lib, stdenv, appimageTools, desktop-file-utils, fetchurl }:
```
Comment (viraptor) [inline pkgs/applications/office/p3x-onenote/default.nix:1]:
```suggestion
{ lib, stdenvNoCC, appimageTools, desktop-file-utils, fetchurl }:
```

File: pkgs/applications/office/p3x-onenote/default.nix [hunk 2]
```diff
@@ -1,23 +1,19 @@
...
+  version = "2022.4.127";
   name = "p3x-onenote-${version}";
 
   plat = {
     aarch64-linux = "-arm64";
     armv7l-linux = "-armv7l";
-    i386-linux = "-i386";
-    i686-linux = "-i386";
     x86_64-linux = "";
   }.${stdenv.hostPlatform.system};
```
Comment (viraptor) [inline pkgs/applications/office/p3x-onenote/default.nix:11]:
```suggestion
  }.${stdenvNoCC.hostPlatform.system};
```

File: pkgs/applications/office/p3x-onenote/default.nix [hunk 3]
```diff
@@ -1,23 +1,19 @@
...
   sha256 = {
-    aarch64-linux = "0a3c0w1312l6k2jvn7cn8priibnh8wg0184zjcli29f9ds1afl5s";
-    armv7l-linux = "172m2d94zzm8q61pvnjy01cl5fg11ad9hfh1han0gycnv3difniy";
-    i386-linux = "12m0i5sb15sbysp5fvhbj4k36950m7kpjr12n88r5fpkyh13ihsp";
-    i686-linux = "12m0i5sb15sbysp5fvhbj4k36950m7kpjr12n88r5fpkyh13ihsp";
-    x86_64-linux = "0bn48r55l5dh8zcf8ijh3z6hlyp3s6fvfyqc1csvnslm63dfkzcq";
+    aarch64-linux = "09sahxil3lcrr40l1fawglmafl643f5anyi9rlhgi2461yghm4qj";
+    armv7l-linux = "14kf455z8705k7ixys4b6rsf1jzsyws6b3gll0bi2ziig0drzzxl";
+    x86_64-linux = "05zr0b06qrlg0kbg58l7wn4jn79jcmzxysm6m30qfmkzliyc33sr";
   }.${stdenv.hostPlatform.system};
```
Comment (viraptor) [inline pkgs/applications/office/p3x-onenote/default.nix:17]:
```suggestion
  }.${stdenvNoCC.hostPlatform.system};
```

**Review by viraptor — COMMENTED** [review]
Since you're not building anything, `stdenvNoCC` saves downloading unnecessary extras during review.

The app doesn't start up correctly for me however. I've got a black window and the following errors in the console:
```
Gtk-Message: 20:44:29.450: Failed to load module "colorreload-gtk-module"
```

```
[2:0829/204429.505035:ERROR:platform_font_skia.cc(246)] Could not find any font: Noto Sans, sans. Falling back to the default
```

```
Error sending from webFrameMain:  Error: Render frame was disposed before WebFrameMain could be accessed
    at n.send (node:electron/js2c/browser_init:165:417)
    at b.send (node:electron/js2c/browser_init:161:2494)
    at BrowserWindow.<anonymous> (/nix/store/1ydwylim95ny60h2gmgq4vg8c1l2w343-p3x-onenote-2022.4.127-extracted/resources/app.asar/src/electron/main/create/window/onenote.js:66:55)
    at BrowserWindow.emit (node:events:538:35)
[29:0829/204434.537527:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
```

Don't know if any are related to the broken display?
I'm using `github:guibou/nixGL#nixGLIntel` to get the gl environment prepared.

**Comment by LegitMagic** [comment]
> The app doesn't start up correctly for me however. I've got a black window and the following errors in the console...
From my usage of it on NixOS it seems to work fine. Do other electron applications work on your machine through nixGL? (also you might as well test with [2022.10.109](https://github.com/patrikx3/onenote/releases/tag/v2022.10.109) while you're at it.)

**Comment by viraptor** [comment]
Other electron apps work fine (discord for example).
The previous version of p3x-onenote crashed hard in the app itself:
```
(lots of no-name internal p3x-onenote frames)
#33 0x0000562f5ed5d207 _ZN2v88internal9Execution4CallEPNS0_7IsolateENS0_6HandleINS0_6ObjectEEES6_iPS6_ (p3x-onenote + 0x2612207)
#34 0x0000562f5ec5d6de _ZN2v88Function4CallENS_5LocalINS_7ContextEEENS1_INS_5ValueEEEiPS5_ (p3x-onenote + 0x25126de)
...
```

**Comment by tiagolobocastro** [comment]
I've updated to stdenvNoCC.
Shall we raise a separate issue for the app not starting as it's perhaps unrelated to this PR?

**Comment by viraptor** [comment]
Yeah, if it works for others, I'm happy with that

**Pull request closed without merging**